### PR TITLE
Static config for setting up smartstore

### PR DESCRIPTION
**Pull request by wmathurin**
Two new methods on SalesforceSDKManagerWithSmartStore:
* `setupGlobalStoreFromDefaultConfig`
Call at any time to register soups defined in resource file`globalstore.json` into the default global smartstore.

* `setupUserStoreFromDefaultConfig`
Call once a user has logged in to register soups defined in resource file `userstore.json` into the user's default smartstore.

Notes:
* The JSON format is based on what we use in our Cordova plugin and React Native module.
* It just does a bunch of `registerSoup` calls, it will not do an `alterSoup` if the soup already exists.
* The only feedback is through the logs - that's intentional, those config files are static (can't be changed at runtime) so the developer should get them right while developing the app.
* You can't define a soup that defines external storage through the config file - that's intentional, we want to phase that feature out at some point.

The sample application SmartSyncExplorer now defines its soup in a config file.
I also added a new test suite called SFSDKStoreConfigTests.

Example of JSON config file
```json
{
  "soups": [
    {
      "soupName": "soup1",
      "indexes": [
        { "path": "stringField1", "type": "string"},
        { "path": "integerField1", "type": "integer"},
        { "path": "floatingField1", "type": "floating"},
        { "path": "json1Field1", "type": "json1"},
        { "path": "ftsField1", "type": "full_text"}
      ]
    }
  ]
}
```

### CODE REVIEW

File: libs/SmartStore/SmartStore/Classes/SFSDKStoreConfig.m [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import <SalesforceSDKCore/SFJsonUtils.h>
```
Comment (bhariharan) [inline libs/SmartStore/SmartStore/Classes/SFSDKStoreConfig.m]: I don't think we need this import if we're importing the umbrella header below.

File: libs/SmartStore/SmartStoreTestApp/globalstore.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (bhariharan) [inline libs/SmartStore/SmartStoreTestApp/globalstore.json]: Should this be under the `Resources` folder in the filesystem and workspace?

File: libs/SmartStore/SmartStoreTestApp/userstore.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (bhariharan) [inline libs/SmartStore/SmartStoreTestApp/userstore.json]: Should this be under the `Resources` folder in the filesystem and workspace?

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/userstore.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/userstore.json]: Should this be under the `Resources` folder in the filesystem and workspace?

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 55%</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

## Current coverage for SalesforceAnalytics.framework is `43.55%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `15.29%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartStore.framework is `62.59%`
Files changed | - | - 
--- | --- | ---
SalesforceSDKManagerWithSmartStore.m | `75.00%` | :warning:
SFSDKStoreConfig.m | `84.85%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by wmathurin** [comment]
Some projects have a `Resources` group (that you see in XCode) but most don't have an actual `Resources` folder on the filesystem. (Actually there is only one such folder: `./libs/SalesforceAnalytics/Resources`).

**Pull request merged**